src/sentry/static/sentry/app/components/selectMembers/index.tsx

```diff
@@ -0,0 +1,359 @@
+          >
+            {this.renderUserBadge(user)}
+          </Tooltip>
+        </DisabledLabel>
+      ),
+    };
+  };
+
+  createMentionableTeam = (team: Team): MentionableTeam => {
+    return {
+      value: team.id,
+      label: <IdBadge team={team} />,
+      searchKey: `#${team.slug}`,
+      actor: {
+        type: 'team',
+        id: team.id,
+        name: team.slug,
+      },
+    };
+  };
+
+  createUnmentionableTeam = (team: Team): UnmentionableTeam => {
+    const {organization} = this.props;
+    const canAddTeam = organization.access.includes('project:write');
+
+    return {
+      ...this.createMentionableTeam(team),
+      disabled: true,
+      label: (
+        <UnmentionableTeam>
+          <DisabledLabel>
+            <Tooltip
+              position="left"
+              title={t('%s is not a member of project', `#${team.slug}`)}
+            >
+              <IdBadge team={team} />
+            </Tooltip>
+          </DisabledLabel>
+          <Tooltip
+            title={
+              canAddTeam
+                ? t('Add %s to project', `#${team.slug}`)
+                : t('You do not have permission to add team to project.')
+            }
+          >
+            <AddToProjectButton
+              size="zero"
+              borderless
+              disabled={!canAddTeam}
+              onClick={this.handleAddTeamToProject.bind(this, team)}
+            >
+              <InlineSvg src="icon-circle-add" />
+            </AddToProjectButton>
+          </Tooltip>
+        </UnmentionableTeam>
+      ),
+    };
+  };
+
+  getMentionableUsers() {
+    return MemberListStore.getAll().map(this.createMentionableUser);
+  }
+
+  getMentionableTeams(): MentionableTeam[] {
+    const {project} = this.props;
+    const projectData = project && ProjectsStore.getBySlug(project.slug);
+
+    if (!projectData) {
+      return [];
+    }
+
+    return projectData.teams.map(this.createMentionableTeam);
+  }
+
+  /**
+   * Get list of teams that are not in the current project, for use in `MultiSelectMenu`
+   *
+   * @param {Team[]} teamsInProject A list of teams that are in the current project
+   */
+  getTeamsNotInProject(teamsInProject: MentionableTeam[] = []): UnmentionableTeam[] {
+    const teams: Team[] = TeamStore.getAll() || [];
+    const excludedTeamIds = teamsInProject.map(({actor}) => actor.id);
+
+    return teams
+      .filter(team => excludedTeamIds.indexOf(team.id) === -1)
+      .map(this.createUnmentionableTeam);
+  }
+
+  /**
+   * Closes the select menu by blurring input if possible since that seems to be the only
+   * way to close it.
+   */
+  closeSelectMenu() {
+    if (!this.selectRef.current) {
+      return;
+    }
+
+    const select = this.selectRef.current.select;
+    const input: HTMLInputElement = select.input.input;
+    if (input) {
+      // I don't think there's another way to close `react-select`
+      input.blur();
```

Comment:
Interesting 🤔

Reply:
this is mostly C&P from an existing component